### PR TITLE
fix-image-tag-behavior

### LINKED ISSUE #203: Clicking on image tags not working as expected
-> Expected Behavior :- When we click on image tag, we must get search results for the respective tag.

-> Current status:- When we click on the image tag the URL in the URL bar changes accordingly but the search results are not updated.

-> For instance when I click on tag "nature" from some image of "city" we get following:-  

screenshot 55

### PR DESCRIPTION
**Pull request by VatsalSin**
The image tags when clicked render the expected results.

Fixes #203

### CODE REVIEW

File: src/components/PhotoDetails.vue [hunk 1]
```diff
@@ -147,10 +147,12 @@ export default {
         if (image.creator) {
           if (image.creator_url) {
             byCreator = `by <a href="${image.creator_url}">${image.creator}</a>`;
-          } else {
+          }
```
Comment (dhruvkb) [inline src/components/PhotoDetails.vue]: These changes are not required and can potentially introduce merge conflicts in other pending PRs that change this file.

File: src/components/PhotoDetails.vue [hunk 2]
```diff
@@ -147,10 +147,12 @@ export default {
         if (image.creator) {
           if (image.creator_url) {
             byCreator = `by <a href="${image.creator_url}">${image.creator}</a>`;
-          } else {
+          } 
+          else {
             byCreator = `by ${image.creator}`;
           }
-        } else {
+        }
```
Comment (dhruvkb) [inline src/components/PhotoDetails.vue]: The same applies to this change as well.

File: src/components/PhotoDetails.vue (hunk 1 above)
Comment (VatsalSin) [reply src/components/PhotoDetails.vue]: I did this to supress he eslint warnings. I will revert this back.

File: src/components/PhotoTags.vue [hunk 3]
```diff
@@ -41,6 +41,7 @@ export default {
     },
     searchByTagName(query) {
       this.$router.push({ name: 'browse-page', query: { q: query } });
+      this.$router.go(0);
```
Comment (brenoferreira) [inline src/components/PhotoTags.vue]: Seems to me a bit strange to add this. Are you sure this is necessary? It might be something else causing this bug

Comment (brenoferreira) [reply src/components/PhotoTags.vue]: [In other places](https://github.com/creativecommons/cccatalog-frontend/blob/master/src/components/SearchGridForm.vue#L122), we use the store to commit the `SET_QUERY` mutation. I think this is the way to go about fixing this bug here

Comment (VatsalSin) [reply src/components/PhotoTags.vue]: I have made the changes using SET_QUERY mutation. Thanks for guiding me in this :-)

**Comment by alishamohanty** [comment]
@VatsalSin It would be better if you could squash your commits into one, since the changes are minor. :)

**Comment by brenoferreira** [comment]
@VatsalSin if you merge this with the changes in #234 that are now in `master`, I think this should be good to go too

**Pull request merged**